File: Data Structures/Structures/Queue/queue.py
class queue:
    # constructor
    def __init__(self, size):
        # sets pointers to -1
        self.__front = -1
        self.__rear = -1
        # sets size var to size given by user
        self.__size = size
        # creates array of size given by user
        self.__data = [None] * self.__size
    
    # checks if the queue is full
    def isFull(self):
        return (self.__rear + 1) % self.__size == self.__front
    
    # checks if the queue is empty
    def isEmpty(self):
        return (self.__front == -1)
    
    # returns the item at the front
    def front(self):
        # checks if the queue is not empty
        if not queue.isEmpty(self):
            # returns the top item
            return self.__data[self.front]
        # returns None if no items are in the queue
        return None

    # adds an item to the queue
    def enqueue(self, item):
        # id the queue has space
        if not queue.isFull(self):
            # if the queue is empty
            if (queue.isEmpty(self)):
                self.__front = 0
            # increment rear
            self.__rear = (self.__rear + 1) % 5
            # add data at the back of the queue
            self.__data[self.__rear] = item
            # return true
            return True
        # return false if item was not added
        return False

    # method to remove an item form the queue
    def dequeue(self):
        # checks if there is an item to remove
        if not queue.isEmpty(self):
            # gets the item
            item = self.__data[self.__front]
            # check if the queue is empty now
            if self.__front == self.__rear:
                # reset pointers if empty
                self.__rear == -1
                self.__front == -1
            else:
                # increment front if not empty
                self.__front = (self.__front + 1) % self.__size
            # returns the item
            return item
        # returns None if queue was empty
        return None

# tests
s = queue(5)
s.enqueue(1)
s.enqueue(2)
s.enqueue(3)
s.enqueue(4)
s.enqueue(5)
print(s.dequeue())
s.enqueue(6)
print(s.dequeue())
print(s.dequeue())
print(s.dequeue())
print(s.dequeue())
print(s.dequeue())

File: Data Structures/Data Structures/Structures/Queue/Queue.java
package Structures.Queue;

public class Queue<E> implements IQueue<E> {
    // holds size of queue
    private int size;
    // holds data for the queue
    private Object[] arr;
    // front of queue
    private int front;
    // back of queue
    private int rear;
    
    /**
     * Constructor method for the queue. 
     *
     * @param s   the size
     */
    public Queue(int s) {
        // sets top to -1 & creates an array of size given by user
        this.size = s;
        this.arr = new Object[s];
        // init front and rear to -1
        this.rear = -1;
        this.front = -1;
    }

    /**
     * Method to check if the queue is full. 
     *
     * @return      whether the queue is full
     */
    public boolean isFull() {
        // if the next item will be placed at the front
        return front == (rear + 1) % size;
    }

    /**
     * Method to check if the queue is empty. 
     *
     * @return      whether the queue is empty
     */
    public boolean isEmpty() {
        // if the front var has not been initialised
        return front == -1;
    }

    /**
     * Method to add an item to the queue. 
     *
     * @param   e   element to add to the queue 
     * @return      whether the item could be added or not
     */
    public boolean enqueue(E e) {
        // if the queue is not full
        if (!isFull()) {
            // init front to 0 if it is at -1
            if (front == -1) {
                front = 0;
            }
            // increment rear
            rear = (rear + 1) % size;
            // add item in back of queue
            arr[rear] = e;
            // return true
            return true;
        }
        // return false if queue is full
        return false;
    }

    /**
     * Method to remove an item from the front of the queue. 
     *
     * @return      whether the item could be removed or not
     */
    public E dequeue() {
        // if the queue is not empty
        if (!isEmpty()) {
            // get the front item
            E item = (E) arr[front];
            // check if the queue is now empty
            if (front == rear) {
                // reset front and rear to -1
                front = -1;
                rear = -1;
            } else {
                // increment the front pointer
                front = (front + 1) % size;
            }
            // return the item
            return item;
        }
        // return null if an item was not found
        return null;
    }

    public void printq() {
        for (Object i: arr) {
            System.out.print(i);
            System.out.print(" ");
        }
        System.out.print('\n');
    }

    /**
     * Method to return the item at the front of the queue. 
     *
     * @return      the front item
     */
    public E front() {
        // checks if there are any items in the queue
        if (!isEmpty()) {
            // returns the front item
            return (E) arr[front];
        }
        // returns null if no item was found
        return null;
    }

    // tests
    public static void main(String[] args) {
        Queue s = new Queue<>(5);
        System.out.println(s.enqueue(1));
        System.out.println(s.enqueue(2));
        System.out.println(s.enqueue(3));
        System.out.println(s.enqueue(4));
        System.out.println(s.enqueue(5));
        System.out.println(s.dequeue());
        System.out.println(s.enqueue(6));
        s.printq();
        System.out.println(s.dequeue());
        System.out.println(s.dequeue());
        System.out.println(s.dequeue());
        System.out.println(s.dequeue());
        System.out.println(s.dequeue());

    }
}
